pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/analytics/PlaybackSessionManager.kt

```diff
@@ -221,60 +243,61 @@ class PlaybackSessionManager {
-                    if (session == currentSession) {
-                        currentSession = null
+                    if (session == _currentSession) {
+                        setCurrentSession(null, eventTime.currentPlaybackPositionMs)
                     } else {
-                        notifyListeners { onSessionFinished(session) }
+                        notifyListeners { onSessionDestroyed(session) }
                         sessions.remove(session.periodUid)
                     }
                 }
             }
         }
 
         override fun onLoadStarted(
-            eventTime: AnalyticsListener.EventTime,
+            eventTime: EventTime,
             loadEventInfo: LoadEventInfo,
             mediaLoadData: MediaLoadData,
         ) {
             getOrCreateSession(eventTime)
         }
 
         override fun onPlayerError(
-            eventTime: AnalyticsListener.EventTime,
+            eventTime: EventTime,
             error: PlaybackException,
         ) {
             DebugLogger.debug(TAG, "onPlayerError", error)
             getOrCreateSession(eventTime)
         }
-        override fun onPlayerReleased(eventTime: AnalyticsListener.EventTime) {
+
+        override fun onPlayerReleased(eventTime: EventTime) {
             DebugLogger.debug(TAG, "onPlayerReleased")
-            finishAllSessions()
+            finishAllSessions(eventTime)
             listeners.clear()
         }
 
-        override fun onPlaybackStateChanged(eventTime: AnalyticsListener.EventTime, state: Int) {
+        override fun onPlaybackStateChanged(eventTime: EventTime, state: Int) {
             DebugLogger.debug(TAG, "onPlaybackStateChanged ${StringUtil.playerStateString(state)}")
             if (state == Player.STATE_IDLE) {
-                finishAllSessions()
+                finishAllSessions(eventTime)
             } else {
                 getOrCreateSession(eventTime)
             }
         }
 
-        private fun getOrCreateSession(eventTime: AnalyticsListener.EventTime): Session? {
+        private fun getOrCreateSession(eventTime: EventTime): Session? {
             if (eventTime.timeline.isEmpty) {
                 return null
             }
             eventTime.timeline.getWindow(eventTime.windowIndex, window)
-            val periodUid = eventTime.timeline.getUidOfPeriod(window.firstPeriodIndex)
+            val periodUid = eventTime.getUidOfPeriod(window)
             var session = getSessionFromPeriodUid(periodUid)
             if (session == null) {
                 val newSession = Session(periodUid, window.mediaItem)
                 sessions[periodUid] = newSession
                 notifyListeners { onSessionCreated(newSession) }
 
-                if (currentSession == null) {
-                    currentSession = newSession
+                if (_currentSession == null) {
+                    setCurrentSession(SessionInfo(newSession, eventTime.currentPlaybackPositionMs), eventTime.currentPlaybackPositionMs)
```

Comment:
The same position is used for both sessions?

Reply:
Yes because in this case only the `newSession` matters because `_curentSession` is null. So I think we can set the `oldPosition` to null.